File: VoxAnvil/src/main/java/com/qualityplus/anvil/util/AnvilPlaceholderUtils.java
package com.qualityplus.anvil.util;

import com.qualityplus.assistant.api.util.IPlaceholder;
import com.qualityplus.assistant.util.placeholder.Placeholder;
import com.qualityplus.assistant.util.placeholder.PlaceholderBuilder;
import com.qualityplus.enchanting.api.enchantment.ICoreEnchantment;
import lombok.experimental.UtilityClass;

import java.util.List;

@UtilityClass
public class AnvilPlaceholderUtils {

    public PlaceholderBuilder getEnchantBuilder(ICoreEnchantment enchantment) {
        return PlaceholderBuilder.create(
                new Placeholder("enchanting_enchant_displayname", enchantment.getName()),
                new Placeholder("enchanting_enchant_description", enchantment.getDescription()),
                new Placeholder("enchanting_enchant_required_bookshelf", enchantment.getRequiredBookShelf())
        );
    }

    public List<IPlaceholder> getEnchantPlaceholders(ICoreEnchantment enchantment) {
        return getEnchantBuilder(enchantment).get();
    }

    private String emptyIfNull(String input) {
        return input == null || input.equals("") ? "&cEmpty" : input;
    }
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/util/AnvilPlaceholderUtils.kt
package com.qualityplus.anvil.util

import com.qualityplus.assistant.api.util.IPlaceholder
import com.qualityplus.assistant.util.placeholder.Placeholder
import com.qualityplus.assistant.util.placeholder.PlaceholderBuilder
import com.qualityplus.enchanting.api.enchantment.ICoreEnchantment
import lombok.experimental.UtilityClass

@UtilityClass
class AnvilPlaceholderUtils {
    fun getEnchantBuilder(enchantment: ICoreEnchantment): PlaceholderBuilder {
        return PlaceholderBuilder.create(
            Placeholder("enchanting_enchant_displayname", enchantment.name),
            Placeholder("enchanting_enchant_description", enchantment.description),
            Placeholder("enchanting_enchant_required_bookshelf", enchantment.requiredBookShelf)
        )
    }

    fun getEnchantPlaceholders(enchantment: ICoreEnchantment): List<IPlaceholder> {
        return getEnchantBuilder(enchantment).get()
    }

    private fun emptyIfNull(input: String?): String {
        return if (input == null || input == "") "&cEmpty" else input
    }
}